scipy/sparse/linalg/tests/test_special_sparse_arrays.py

```diff
@@ -129,3 +129,8 @@ def test_dot(self, grid_shape, bc, dtype):
             y = lap.dot(x.astype(dtype))
             assert x.shape == y.shape
             assert y.dtype == dtype
+
+def test_boundary_conditions_value_error():
+    with pytest.raises(ValueError) as excinfo:
+        LaplacianNd(grid_shape=(6, ), boundary_conditions="Dirichlet")
+    assert str(excinfo.value).startswith("Unknown value 'Dirichlet'")
```

Comment:
Here is the more compact way to check the error message. Also would prefer a more obviously wrong input as suggested below:

```suggestion
    with pytest.raises(ValueError, match="Unknown value 'robin'"):
        LaplacianNd(grid_shape=(6, ), boundary_conditions="robin")
```